src/runtime/metal.cpp

```diff
@@ -39,13 +39,17 @@ WEAK mtl_buffer *new_buffer(mtl_device *device, size_t length) {
 }
 
 WEAK mtl_command_queue *new_command_queue(mtl_device *device) {
-    return (mtl_command_queue *)objc_msgSend(device, sel_getUid("newCommandQueue"));
+    typedef mtl_command_queue *(*new_command_queue_method)(objc_id dev, objc_sel sel);
+    new_command_queue_method method = (new_command_queue_method)&objc_msgSend;
+    return (mtl_command_queue *)(*method)(device, sel_getUid("newCommandQueue"));
```

Comment:
Do you need to dereference the function pointer? I thought that was a no-op. Also after the function pointer cast, isn't the return type already correct?

```
    return method(device, sel_getUid("newCommandQueue"));
```

Also please add a comment explaining why the cast is necessary.

Reply:
Will add a comment about casting the `objc_msgSend` calls.

For the function pointer deref and the return type cast, I followed what the other functions in `metal.cpp` were doing (I think written by @zvookin?), just to make the changes minimal.  Happy to go back and change them all to not do either of those things, if that's preferable.

Reply:
Favoring consistency with existing code seems reasonable to me.